File: data/multilang_comparison/program_go/program_go/src/func44.go
package main
import "math/rand"
var _ = rand.Intn
func Func44(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var array32 *Array
      if pCounter > 0 {
          array32 = vars.Data[pCounter-1]
          pCounter--
          array32.RefC++
      } else {
          array32 = &Array{
              Size: 440,
              RefC: 1,
              Id: 32,
              Data: make([]uint32, 440),
          }
      }
      var loop12 int = 0;
      var loopLimit12 int = (50)/4 + 1;
      for ; loop12 < loopLimit12; loop12++ {
         for i := 0; i < array32.Size; i++ {
             array32.Data[i]--
         }
         for i := 0; i < array32.Size; i++ {
             if array32.Data[i] == 17 {
                 return array32
             }
         }
      }
      array32.RefC--
      if array32.RefC == 0 {
          array32.Data = nil
          array32 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array33 *Array = Func50(&params0, loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array33.RefC--
      if array33.RefC == 0 {
          array33.Data = nil
          array33 = nil
      }
   }
   var array34 *Array
   if pCounter > 0 {
       array34 = vars.Data[pCounter-1]
       pCounter--
       array34.RefC++
   } else {
       array34 = &Array{
           Size: 481,
           RefC: 1,
           Id: 34,
           Data: make([]uint32, 481),
       }
   }
   return array34;
}

File: data/multilang_comparison/program_c/src/func44.c
#include "program_c.h" 
array_t* func44(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t* array32;
      if (pCounter > 0) {
         array32 = vars->data[--pCounter];
         array32->refC++;
         DEBUG_COPY(array32->id);
      } else {
         array32 = (array_t*)malloc(sizeof(array_t));
         array32->size = 440;
         array32->refC = 1;
         array32->id = 32;
         array32->data = (unsigned int*)malloc(array32->size*sizeof(unsigned int));
         memset(array32->data, 0, array32->size*sizeof(unsigned int));
         DEBUG_NEW(array32->id);
      }
      unsigned int loop12 = 0;
      unsigned int loopLimit12 = (rand()%loopsFactor)/4 + 1;
      for(; loop12 < loopLimit12; loop12++) {
         for (int i = 0; i < array32->size; i++) {
            array32->data[i]--;
         }
         for (int i = 0; i < array32->size; i++) {
            if (array32->data[i] == 17) { 
               return array32;
            }
         }
      }
      array32->refC--;
      if(array32->refC == 0) {
         free(array32->data);
         free(array32);
         DEBUG_FREE(array32->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array33 = func50(&params0, loopsFactor);
      DEBUG_RETURN(array33->id);
      free(params0.data);
      array33->refC--;
      if(array33->refC == 0) {
         free(array33->data);
         free(array33);
         DEBUG_FREE(array33->id);
      }
   }
   array_t* array34;
   if (pCounter > 0) {
      array34 = vars->data[--pCounter];
      array34->refC++;
      DEBUG_COPY(array34->id);
   } else {
      array34 = (array_t*)malloc(sizeof(array_t));
      array34->size = 481;
      array34->refC = 1;
      array34->id = 34;
      array34->data = (unsigned int*)malloc(array34->size*sizeof(unsigned int));
      memset(array34->data, 0, array34->size*sizeof(unsigned int));
      DEBUG_NEW(array34->id);
   }
   return array34;
}

